tools/analyse-storage/src/main.rs

```diff
@@ -0,0 +1,268 @@
+                    // Count key sizes
+                    let key_len = tuple.0.len();
+                    *local_key_sizes.entry(key_len).or_insert(0) += 1;
+
+                    // Count value sizes
+                    let value_len = tuple.1.len();
+                    *local_value_sizes.entry(value_len).or_insert(0) += 1;
+                }
+                Err(err) => {
+                    panic!("Error occured during iteration of {}: {}", col_family, err);
+                }
+            }
+        }
+        println!(
+            "In column family {} there are {} number of pairs, and col size is {}",
+            col_family,
+            local_key_sizes.values().sum::<usize>(),
+            local_key_sizes.iter().map(|(&size, &count)| size * count).sum::<usize>()
+        );
+        update_map(&key_sizes, &local_key_sizes);
+        update_map(&value_sizes, &local_value_sizes);
+    });
+
+    let mut key_sizes_sorted: Vec<(usize, usize)> =
+        key_sizes.lock().unwrap().clone().into_iter().collect();
+    key_sizes_sorted.sort_by(|a, b| b.1.cmp(&a.1));
+    let mut value_sizes_sorted: Vec<(usize, usize)> =
+        value_sizes.lock().unwrap().clone().into_iter().collect();
+    value_sizes_sorted.sort_by(|a, b| b.1.cmp(&a.1));
+    (key_sizes_sorted, value_sizes_sorted)
+}
+
+fn get_column_family_options(
+    input_col: Option<String>,
+) -> (Vec<String>, Vec<ColumnFamilyDescriptor>) {
+    match input_col {
+        Some(col_name) => {
+            let mut opts = Options::default();
+            let maybe_db_col = get_db_col(&col_name);
+            if let Some(db_col) = maybe_db_col {
+                if db_col.is_rc() {
+                    opts.set_merge_operator(
+                        "refcount merge",
+                        RocksDB::refcount_merge,
+                        RocksDB::refcount_merge,
+                    );
+                    opts.set_compaction_filter(
+                        "empty value filter",
+                        RocksDB::empty_value_compaction_filter,
+                    );
+                }
+            }
+            (vec![col_name.clone()], vec![ColumnFamilyDescriptor::new(col_name, opts)])
+        }
+        None => {
+            let col_families = get_all_col_familiy_names();
+            (
+                col_families.iter().map(|db_col| col_name(*db_col).to_string()).collect(),
+                col_families
+                    .iter()
+                    .map(|db_col| {
+                        ColumnFamilyDescriptor::new(
+                            col_name(*db_col),
+                            rocksdb_column_options(*db_col),
+                        )
+                    })
+                    .collect(),
+            )
+        }
+    }
+}
+
+fn main() {
+    let args = Cli::parse();
+
+    // Set db options
+    let mut opts = Options::default();
+    opts.set_max_open_files(20_000);
+    opts.increase_parallelism(std::cmp::max(1, num_cpus::get() as i32 / 2));
```

Comment:
Hm but why we use opts different from `rocksdb_options(...)`?

Reply:
Because we are not interested in writing to RocksDB not reading random values, only in iterating through all values, so the idea was just to speed that up

Reply:
Let's put this explanation to the comment then, it's much comprehensive for the reader who doesn't have any context than "Set db options" :)